lgc/builder/ImageBuilder.cpp

```diff
@@ -730,7 +731,68 @@ Value *ImageBuilder::CreateImageSample(Type *resultTy, unsigned dim, unsigned fl
 Value *ImageBuilder::CreateImageSampleConvert(Type *resultTy, unsigned dim, unsigned flags, Value *imageDesc,
                                               Value *convertingSamplerDesc, ArrayRef<Value *> address,
                                               const Twine &instName) {
-  llvm_unreachable("Not yet implemented");
+  return CreateImageSampleConvertYCbCr(resultTy, dim, flags, imageDesc, convertingSamplerDesc, address, instName);
+}
+
+// =====================================================================================================================
+// Create an image sample with a converting sampler.
+// The caller supplies all arguments to the image sample op in "address", in the order specified
+// by the indices defined as ImageAddressIdx* below.
+//
+// @param resultTy : Result type
+// @param dim : Image dimension
+// @param flags : ImageFlag* flags
+// @param imageDesc : Image descriptor
+// @param convertingSamplerDesc : Converting sampler descriptor (v8i32)
+// @param address : Address and other arguments
+// @param instName : Name to give instruction(s)
+Value *ImageBuilder::CreateImageSampleConvertYCbCr(Type *resultTy, unsigned dim, unsigned flags, Value *imageDesc,
+                                                   Value *convertingSamplerDesc, ArrayRef<Value *> address,
+                                                   const Twine &instName) {
+  Value *pResult = nullptr;
+  Value *ycbcrSamplerDesc = CreateExtractValue(convertingSamplerDesc, 0);
+
+  // Extract YCbCr meta data
+  const SamplerYCbCrConversionMetaData yCbCrMetaData = {
+      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue()),
+      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(5)))->getZExtValue()),
+      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(6)))->getZExtValue()),
+      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(7)))->getZExtValue()),
+  };
+
+  // Prepare the coordinate and derivatives, which might also change the dimension.
+  SmallVector<Value *, 4> coords;
+  SmallVector<Value *, 6> derivatives;
+
+  Value *projective = address[ImageAddressIdxProjective];
+  if (projective != nullptr) {
+    projective = CreateFDiv(ConstantFP::get(projective->getType(), 1.0), projective);
```

Comment:
According to new coding style:
Single-line statement needn't {}
if (projective != nullptr)  -> if (projective)